Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -277,7 +278,19 @@ static void softlinkDataDetectorsFrameworks()
 
 #if HAVE(VIDEO_RESTRICTED_DECODING)
 #if PLATFORM(MAC)
-    SandboxExtension::consumePermanently(parameters.trustdExtensionHandle);
+    if (SandboxExtension::consumePermanently(parameters.trustdExtensionHandle)) {
+        // Open up a Mach connection to trustd by doing a code check validation.
+        // This is required since launchd will be blocked after process launch, which prevents new Mach connections to be created.
+        // The actual path is not important in this case, since we just want to open a Mach connection.
```

Comment:
Are we sure that opening a Mach connection will not violate the sandbox itself? It is just weird that we apply the sandbox but we have to allow frameworks outside it such as trustd and launchd.